### ISSUE TITLE
Various Errors Relating to Expected Adjacents

### ISSUE THREAD

**Issue by lukeduran** [RadiantLabs/modeling-api-docs#50]
We're receiving several variations of errors relating to Expected Adjacents. Request and response is below. Please note that the errors array itself is quite long and is not well formatted below.

```{
    "request": {
        "headers": {
            "User-Agent": "python-requests/2.28.1",
            "Accept-Encoding": "gzip, deflate",
            "Accept": "*/*",
            "Connection": "keep-alive",
            "Content-Type": "application/json",
            "Authorization": "Bearer jqnTR0wR6nmnS6u5mLEm1NSXq1kHSaF16LYlInbQ8tC04QHa",
            "Content-Length": "1412"
        },
        "body": {
            "addressFull": "218 S Ohio St, King City, MO 64463-9530",
            "addressComponents": {
                "addressLine1": "218 S Ohio St",
                "city": "King City",
                "stateAbbreviation": "MO",
                "zipCode": "64463-9530"
            },
            "baseBuilding": {
                "buildingSummary": {
                    "yearBuilt": 1929,
                    "bedroomsCount": 2,
                    "averageCeilingHeight": 8,
                    "conditionedFloorArea": 1077
                },
                "enclosure": {
                    "airInfiltration": {
                        "rate": 2313,
                        "rateUnit": "CFM"
                    },
                    "attics": [
                        {
                            "floorAssemblyEffectiveRValue": 38,
                            "id": "attic-insulation-1"
                        }
                    ],
                    "foundations": [
                        {
                            "type": "crawl unvented",
                            "id": "floor-and-foundation-insulation-1"
                        }
                    ],
                    "walls": [
                        {
                            "assemblyEffectiveRValue": 5,
                            "id": "wall-insulation-1"
                        }
                    ]
                },
                "systems": {
                    "hvac": {
                        "hvacHeatPumps": [
                            {
                                "systemType": "air-to-air",
                                "heatLoadFraction": 1.0,
                                "coolLoadFraction": 1.0,
                                "heatEfficiency": 9.5,
                                "coolEfficiency": 16.0,
                                "backupSystem": {},
                                "id": "heat-pump-systems-1"
                            }
                        ]
                    },
                    "waterHeating": {
                        "waterHeatingSystems": [
                            {
                                "fuel": "electricity",
                                "dhwLoadFraction": 1.0,
                                "id": "water-heating-1"
                            }
                        ]
                    }
                }
            },
            "globalControls": {
                "debug": true,
                "includeInputs": true,
                "includeBase": true,
                "includeTypical": true,
                "useFixture": false,
                "resolution": [
                    "annual",
                    "monthly"
                ],
                "timelineStartDate": "2024-02-27",
                "timelineDurationInYears": 1,
                "units": {
                    "energyStandard": "MBtu",
                    "coal": "metric ton",
                    "electricity": "MBtu",
                    "fuelOil": "MBtu",
                    "naturalGas": "MBtu",
                    "propane": "MBtu",
                    "woodCord": "MBtu",
                    "woodPellets": "MBtu",
                    "monetary": "USD",
                    "emissions": "CO2"
                }
            }
        }
    },
    "response": {
        "headers": {
            "Server": "nginx/1.24.0",
            "Date": "Tue, 27 Feb 2024 16:39:59 GMT",
            "Content-Type": "application/json;charset=utf-8",
            "Content-Length": "6229",
            "Connection": "keep-alive",
            "X-Request-Id": "01hqnps0td65cjnnxm1xzzw3kc",
            "Strict-Transport-Security": "max-age=31536000"
        },
        "body": {
            "globalControls": {
                "timelineStartDate": "2024-02-27",
                "debug": true,
                "includeInputs": true,
                "resolution": [
                    "monthly",
                    "annual"
                ],
                "includeBase": true,
                "includeTypical": true,
                "useFixture": false,
                "timelineDurationInYears": 1,
                "units": {
                    "monetary": "USD",
                    "electricity": "MBtu",
                    "naturalGas": "MBtu",
                    "woodPellets": "MBtu",
                    "propane": "MBtu",
                    "coal": "metric ton",
                    "emissions": "CO2",
                    "woodCord": "MBtu",
                    "energyStandard": "MBtu",
                    "fuelOil": "MBtu"
                }
            },
            "addressFull": "218 S Ohio St, King City, MO 64463-9530",
            "baseBuilding": {
                "systems": {
                    "waterHeating": {
                        "waterHeatingSystems": [
                            {
                                "dhwLoadFraction": 1.0,
                                "fuel": "electricity",
                                "id": "water-heating-1"
                            }
                        ]
                    },
                    "hvac": {
                        "hvacHeatPumps": [
                            {
                                "heatLoadFraction": 1.0,
                                "heatEfficiency": 9.5,
                                "backupSystem": {},
                                "coolLoadFraction": 1.0,
                                "coolEfficiency": 16.0,
                                "systemType": "air-to-air",
                                "id": "heat-pump-systems-1"
                            }
                        ]
                    }
                },
                "enclosure": {
                    "airInfiltration": {
                        "rate": 2313,
                        "rateUnit": "CFM"
                    },
                    "walls": [
                        {
                            "assemblyEffectiveRValue": 5,
                            "id": "wall-insulation-1"
                        }
                    ],
                    "foundations": [
                        {
                            "type": "crawl unvented",
                            "id": "floor-and-foundation-insulation-1"
                        }
                    ],
                    "attics": [
                        {
                            "floorAssemblyEffectiveRValue": 38,
                            "id": "attic-insulation-1"
                        }
                    ]
                },
                "buildingSummary": {
                    "conditionedFloorArea": 1077,
                    "yearBuilt": 1929,
                    "bedroomsCount": 2,
                    "averageCeilingHeight": 8
                }
            },
            "addressComponents": {
                "zipCode": "64463-9530",
                "addressLine1": "218 S Ohio St",
                "stateAbbreviation": "MO",
                "city": "King City"
            },
            "errors": [
                {
                    "message": "[\"Warning: No dishwasher specified, the model will not include dishwasher energy use.\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected InteriorAdjacentTo to be 'attic' or 'attic - conditioned' or 'attic - unconditioned' or 'attic - unvented' or 'attic - vented' or 'basement' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace' or 'crawlspace - conditioned' or 'crawlspace - unconditioned' or 'crawlspace - unvented' or 'crawlspace - vented' or 'garage' or 'garage - conditioned' or 'garage - unconditioned' or 'ground' or 'living space' or 'other' or 'other heated space' or 'other housing unit' or 'other housing unit above' or 'other housing unit below' or 'other multifamily buffer space' or 'other non-freezing space' or 'outside' or 'unconditioned space' [context: /HPXML/Building/BuildingDetails/Enclosure/RimJoists/RimJoist]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected InteriorAdjacentTo to be 'attic' or 'attic - conditioned' or 'attic - unconditioned' or 'attic - unvented' or 'attic - vented' or 'basement' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace' or 'crawlspace - conditioned' or 'crawlspace - unconditioned' or 'crawlspace - unvented' or 'crawlspace - vented' or 'garage' or 'garage - conditioned' or 'garage - unconditioned' or 'ground' or 'living space' or 'other' or 'other heated space' or 'other housing unit' or 'other housing unit above' or 'other housing unit below' or 'other multifamily buffer space' or 'other non-freezing space' or 'outside' or 'unconditioned space' [context: /HPXML/Building/BuildingDetails/Enclosure/FoundationWalls/FoundationWall]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected ExteriorAdjacentTo to be 'attic' or 'attic - conditioned' or 'attic - unconditioned' or 'attic - unvented' or 'attic - vented' or 'basement' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace' or 'crawlspace - conditioned' or 'crawlspace - unconditioned' or 'crawlspace - unvented' or 'crawlspace - vented' or 'garage' or 'garage - conditioned' or 'garage - unconditioned' or 'ground' or 'living space' or 'other' or 'other heated space' or 'other housing unit' or 'other housing unit above' or 'other housing unit below' or 'other multifamily buffer space' or 'other non-freezing space' or 'outside' or 'unconditioned space' [context: /HPXML/Building/BuildingDetails/Enclosure/FrameFloors/FrameFloor]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected InteriorAdjacentTo to be 'attic' or 'attic - conditioned' or 'attic - unconditioned' or 'attic - unvented' or 'attic - vented' or 'basement' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace' or 'crawlspace - conditioned' or 'crawlspace - unconditioned' or 'crawlspace - unvented' or 'crawlspace - vented' or 'garage' or 'garage - conditioned' or 'garage - unconditioned' or 'ground' or 'living space' or 'other' or 'other heated space' or 'other housing unit' or 'other housing unit above' or 'other housing unit below' or 'other multifamily buffer space' or 'other non-freezing space' or 'outside' or 'unconditioned space' [context: /HPXML/Building/BuildingDetails/Enclosure/Slabs/Slab]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected InteriorAdjacentTo to be 'living space' or 'attic - vented' or 'attic - unvented' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace - vented' or 'crawlspace - unvented' or 'garage' [context: /HPXML/Building/BuildingDetails/Enclosure/RimJoists/RimJoist]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected InteriorAdjacentTo to be 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace - vented' or 'crawlspace - unvented' or 'garage' [context: /HPXML/Building/BuildingDetails/Enclosure/FoundationWalls/FoundationWall]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected ExteriorAdjacentTo to be 'outside' or 'attic - vented' or 'attic - unvented' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace - vented' or 'crawlspace - unvented' or 'garage' or 'other housing unit' or 'other heated space' or 'other multifamily buffer space' or 'other non-freezing space' [context: /HPXML/Building/BuildingDetails/Enclosure/FrameFloors/FrameFloor]\" \"Error: /tmp/modeling-api/dir2/NORMAL/20240227163613775148518/312158427_BASE.xml: Expected InteriorAdjacentTo to be 'living space' or 'basement - conditioned' or 'basement - unconditioned' or 'crawlspace - vented' or 'crawlspace - unvented' or 'garage' [context: /HPXML/Building/BuildingDetails/Enclosure/Slabs/Slab]\"]",
                    "code": "openstudio-creating-input-unsuccessful"
                }
            ]
        }
    }
}```

**Comment by lukeduran** [comment]
@joshuamzm @kellyatradiantlabs we are still receiving several of these errors. Any updates?

**Comment by kloconnell** [comment]
@lukeduran I just reran this request and got a different error than what you shared above. `Duct location is 'crawlspace - vented' but building does not have this location specified.`. This error will be resolved with the 2.0.0 release.

**Comment by kloconnell** [comment]
This request is now passing in v2.0.0, with the addition of ``systemType`` in the ``waterHeatingSystems`` object.
